### ISSUE TITLE
AJAX results do not display when select has existing option

### ISSUE THREAD

**Issue by futura** [brianvoe/slim-select#276]
Not sure if this is similar to other issues, but I didn't find this exact one.

I want to build an AJAX autocomplete select. I want to be able to preload an existing value into the select, so that if the form is submitted without change, the value will be persisted.

But when I add an ```<option>``` to the select menu, I get errors when trying to display AJAX results.

https://jsfiddle.net/Lq6sf831/

Is there another method to do this? It seems like setData() might be what I want, but I'm not sure how to add that to the ajax function instead of callback().

**Comment by futura** [comment]
Maybe this is related to pull request #270? Any timeline on merging that into the main branch? (Sorry, I don't know how to `yarn add` the develop branch, and I don't know how to access that branch from jsfiddle, so I can't test for my use case.) I'd be glad to help if I can.

**Comment by andrewdang17** [comment]
I have this problem as well. A temporary workaround for this is to add the `allowDeselect: true` option and require the user to click that before searching.

**Comment by futura** [comment]
Thanks @andrewdang17, I just added that to the fiddle, and it does work.

But I'm pretty sure my users won't know to deselect the default option in order to search for a different option, especially since the search box appears automatically on focus (which makes it seem like they can just begin searching). Maybe I can programmatically deselect ...

Then there are cases where users might begin searching and change their minds mid-search (perhaps if they don't find an option they're looking for), but then the original value will not be restored.

Bigger picture, I'm trying to drop jQuery and select2, which depends on it. Select2 allows the behavior I'm describing, but not sure how the internals work.

**Comment by andrewdang17** [comment]
for sure, downgrading to `1.26.1` also worked for me. Haven't run into any issues yet with it.

**Comment by futura** [comment]
Oh, interesting, thanks. So there was a change affecting this in 1.26.2. Hopefully, this can be solved in future versions.

**Comment by dragomano** [comment]
I solved the same problem with restoring lines [204 and 205](https://github.com/brianvoe/slim-select/commit/7b32d6187247765ed7891d8cdd5d6908cdcf1368#diff-4ea0d5be4f44377893d71d5dcfbe3288e830007fbe8a71972999ea8816e0e18eL204-L205)

**Comment by atstockland** [comment]
I have 1.27.0 installed and still have this issue.  @futura 's comment on May 6th sums up exactly what I am after too.  @futura what direction did you go to solve this on your end?

**Comment by futura** [comment]
@atstockland I downgraded to 1.26.1, and it seems to be working. I don't love that solution, but it does the job for now.

**Comment by atstockland** [comment]
Thanks @futura!  @brianvoe what are your thoughts on this?

**Comment by atstockland** [comment]
Downgrading to 1.26.1 also fixed this for me.

**Comment by mtomov** [comment]
I can confirm the exact reported issue, and that downgrading to 1.26.1 fixes the issue. Thank you all for the dig in to the cause and the suggested temporary fix!

**Comment by brianvoe** [comment]
Hey everyone. A new version of slim select was released. I completely rewrote how the search events work. I renamed ajax to search and it takes a return of promise that will all for more control over the options that are shown in the dropdown when searching. https://slimselectjs.com/events#search Check it out and let me know your thoughts. If this is still an issue let me know and ill work to get it fixed. Thanks.
